### PR TITLE
Mosviz slit overlay

### PR DESCRIPTION
**Pull request by javerbukh**
Mosviz slit overlay implementation as a plugin, as well as a development notebook. I still have debugging statements because this is still a WIP. Only blocker remaining is that the `meta` attribute in `CCDData.read` is losing keys between it's instantiation and this plugin accessing it using 
```
image_data = self.app.get_viewer("image-viewer").data()
wcs_image = WCS(image_data[0].meta)
```
I currently have a hardcoded stand-in for this `meta` information (`test_meta` variable), but once `image_data[0].meta` retains its values I can use that instead.

If testing this, the slit appears only when using the 3rd row from the top (Image 2, 2D Spectrum 2, etc.)

Related ticket in glue-astronomy: https://github.com/glue-viz/glue-astronomy/pull/27

### CODE REVIEW

**Comment by javerbukh** [comment]
Slit creation using the `metadata` of a CCDData object now works! However, because the 2d spectra and cutouts do not seem to be linked, not all rows can produce a slit. 

Work left: Snackbar messages still need to be added to let a user know the status of slit creation and slits need to be removed when a different row is selected.

**Comment by javerbukh** [comment]
Snackbar messages added and debug statements removed. Should be in a good state for testing. Slits do not get removed when a different row is selected, I propose this be a separate PR to be done after the deadline. We are currently looking into how hub subscriptions may be creating unnecessary messages that slow down the application, so I do not want to create new hub subscriptions until after we figure that out.

File: jdaviz/configs/mosviz/plugins/parsers.py [hunk 1]
```diff
@@ -179,7 +182,17 @@ def _parse_as_image(path):
 
             unit = hdulist[0].header.get('BUNIT', 'Jy')
 
-        return CCDData.read(path, unit=unit)
+            header = hdulist[0].header.copy()
+            meta = {}
+            for x in header.keys():
+                meta[x] = header[x]
```
Comment (nmearl) [inline jdaviz/configs/mosviz/plugins/parsers.py:188]:
```suggestion
            meta = dict(header)
```

Comment (javerbukh) [reply jdaviz/configs/mosviz/plugins/parsers.py]: Good call!

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <v-card-actions>
+      <div class="flex-grow-1"></div>
+      <v-btn
+      color="primary" text @click="slit_overlay">Apply</v-btn>
+    </v-card-actions>
+
+    <v-card-actions>
+      <div class="flex-grow-1"></div>
+      <v-btn
+      color="primary" text @click="slit_overlay_remove">Remove</v-btn>
```
Comment (rosteen) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.vue:14]:
```suggestion
      <div class="flex-grow-1">
        <v-row align="center" justify="center">
          <v-btn
            color="primary" text @click="slit_overlay">Apply</v-btn>
          <v-btn
            color="primary" text @click="slit_overlay_remove">Remove</v-btn>
        </v-row>
      </div>
```

Comment (javerbukh) [reply jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.vue]: Thanks! Much clearer now.

**Comment by PatrickOgle** [comment]
It appears there is no longer a Plugin and the slit overlay is rendered automatically.   @javerbukh please confirm.

**Review by PatrickOgle — APPROVED** [review]
The slit overlay now appears automatically for each different row.  There is no longer an overlay Plugin.
This is a much cleaner solution and much appreciated.  The aperture moves around to a different location in the image for each row.

**Comment by javerbukh** [comment]
Just to clarify @PatrickOgle's comment, Slit Overlay is still implemented as a plugin, but there is a discussion to make it a plot option with the implementation existing in the mosviz helper class.

**Pull request merged**